### PR TITLE
Add lib content based on modules

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
Initial commits for this repository, exporting the current version of the Faq Client we have on our modules.

- [x] Add Readme
- [x] Move FaqClient from other modules in it
- [x] Allow additional function to be called in case of error
- [x] Add unit tests
- [x] Add GA workflows

### CODE REVIEW

File: src/Faq.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+            if (!$e->hasResponse()) {
+                return false;
+            }
+            $response = $e->getResponse();
+        }
+
+        $data = json_decode($response->getBody(), true);
+
+        return isset($data['categories']) && !empty($data['categories']) ? $data : false;
```
Comment (Quetzacoalt91) [inline src/Faq.php]:
```suggestion
        return !empty($data['categories']) ? $data : false;
```

**Comment by Quetzacoalt91** [comment]
CI result can be found on my fork: https://github.com/Quetzacoalt91/module-lib-faq/actions/runs/306827132

Once this one is merged, all other pull-requests will be able to display the results without having to look into the forks.

File: src/Faq.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    public function __construct($moduleKey, $psVersion, $isoCode, array $options = [])
+    {
+        $this->parameters = (new Parameters())
+            ->setModuleKey($moduleKey)
+            ->setPsVersion($psVersion)
+            ->setIsoCode($isoCode);
+
+        // Allow client options to be customized
+        $options = array_merge_recursive([
+            'base_url' => self::BASE_URL,
```
Comment (Matt75) [inline src/Faq.php:64]: Avoid `cURL error 60: SSL certificate problem: unable to get local issuer certificate` on windows and others server without a cacert or missing php configuration.
```suggestion
            'base_url' => self::BASE_URL,
            'verify' => defined('_PS_CACHE_CA_CERT_FILE_') ? constant('_PS_CACHE_CA_CERT_FILE_') : true,
```

**Comment by Quetzacoalt91** [comment]
Thanks for the review @Matt75

**Pull request merged**